### PR TITLE
Store overview metrics in Redux

### PR DESCRIPTION
**Pull request by spadgett**
Avoid passing metrics down through several layers of components.

/cc @TheRealJon @alecmerdler

### CODE REVIEW

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by alecmerdler** [comment]
jenkins rebuild

File: frontend/public/components/overview/index.tsx [hunk 1]
```diff
@@ -357,17 +356,18 @@ const OverviewHeading_: React.SFC<OverviewHeadingProps> = ({disabled, firstLabel
...
-  const selectedView = UI.getIn(['overview', 'selectedView'], View.Resources);
-  return { selectedView };
+  const metrics = UI.getIn(['overview', 'metrics']);
+  const selectedView = UI.getIn(['overview', 'selectedView']);
+  return { metrics, selectedView };
 };
 
 const mainContentDispatchToProps = (dispatch): OverviewMainContentPropsFromDispatch => ({
+  updateMetrics: (metrics: OverviewMetrics) => dispatch(UIActions.updateOverviewMetrics(metrics)),
   updateResources: (items: any[]) => dispatch(UIActions.updateOverviewResources(items)),
```
Comment (TheRealJon) [inline frontend/public/components/overview/index.tsx]: (not new)

```suggestion
  updateResources: (items: OverviewItem[]) => dispatch(UIActions.updateOverviewResources(items)),
```

File: frontend/public/components/overview/index.tsx [hunk 2]
```diff
@@ -1045,10 +1050,12 @@ type OverviewHeadingOwnProps = {
 type OverviewHeadingProps = OverviewHeadingPropsFromState & OverviewHeadingPropsFromDispatch & OverviewHeadingOwnProps;
 
 type OverviewMainContentPropsFromState = {
+  metrics: OverviewMetrics;
   selectedView: View;
 };
 
 type OverviewMainContentPropsFromDispatch = {
+  updateMetrics: (metrics: OverviewMetrics) => void;
   updateResources: (items: any[]) => void;
```
Comment (TheRealJon) [inline frontend/public/components/overview/index.tsx]: (not new)
```suggestion
  updateResources: (items: OverviewItem[]) => void;
```

File: frontend/public/ui/ui-reducers.js [hunk 3]
```diff
@@ -100,10 +102,14 @@ export default (state, action) => {
     case types.dismissOverviewDetails:
       return state.mergeIn(['overview'], {selectedUID: '', selectedDetailsTab: ''});
 
+    case types.updateOverviewMetrics:
+      return state.mergeIn(['overview', 'metrics'], action.metrics);
```
Comment (TheRealJon) [inline frontend/public/ui/ui-reducers.js]: Why `mergeIn` and not `setIn`? If there are stale metrics stored in this map, wouldn't this cause them not to be removed?

**Comment by spadgett** [comment]
Thanks, updated

**Pull request merged**